libs/sdk/android/src/main/java/com/rudderstack/react/android/RNRudderSdkModule.java

```diff
@@ -249,6 +249,21 @@ public void endSession() {
         RudderLogger.logVerbose("RNRudderSdkModule: endSession: ending session");
     }
 
+    @ReactMethod
+    public void getSessionId(Promise promise) {
+        if (!isRudderClientInitializedAndReady()) {
+            promise.resolve(null);
+            return;
+        }
+        try {
+            String sessionId = (userSessionPlugin.getSessionId() != null) ?
+                    String.valueOf(userSessionPlugin.getSessionId()) : null;
+            promise.resolve(sessionId);
```

Comment:
Why are we returning it as `String` in Java and `NSNumber` in ObjC ?

Can we return as `Double` from Java as well, then we would always get `number?` on the typescript side and we do not need to check for null as well and simplify the code.

Reply:
React Native Android doesn't allow sessionId (or integer) to be returned as `Long`, hence I converted that into `String` and returned it. Whereas in React Native iOS, `NSNumber` is allowed.

IMO, `double` is not an appropriate data type for this case, as it is meant for `decimal-type` values. If you think that I still need to use that LMK.

Reply:
yes, Android->React Native doesn't support `Long` and `Double` is the only numeric type which is supported. 

In that case, do you think returning a String/NSString* itself from both Android and iOS would be a good idea?

Because 
i) when you return String from Android, it would be casted to string when sent to typescript 
ii) and when we return NSNumber from iOS it would be casted as number? when sent to typescript. 

Would that be causing any problems for us ?

Reply:
I understand your concern that we are returning two different types of values: `String` from Android and `NSNumber` from iOS. 
I would like to bring the following things under your consideration:

1. At the React Native side over [here](https://github.com/rudderlabs/rudder-sdk-react-native/blob/46fa845a3b872ccea8344429443ea0a711950e13/libs/sdk/src/RudderClient.ts#L328), I'm converting the returned value explicitly into number before returning to the user.
2. I've tested the current implementation and it's working well and always returning `sessionId` as `number`.

If you still are of the opinion or think it would be better to return NSString from iOS to bring consistency/parity LMK, I would be happy to do that. 
If you have something else please LMK that as well.

Reply:
Awesome, that should be good.